kernels/openvx-core/host/vx_copy_host.c

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright (c) 2012-2024 The Khronos Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+Implementation file for the COPY, SWAP, and MOVE kernels
+*/
+
+#include <TI/tivx.h>
+#include <VX/vx.h>
+#include <vx_internal.h>
```

Comment:
Is this needed still?  I would like to avoid including this unless needed for a specific reason

Reply:
@lcweaver , if I remove this, I get several errors related to the fact that the code is assuming it has access to the underlying structures of references, nodes, etc (uses pointers) whereas the OpenVX API has only opaque references without knowing the underlying structure.

This kernel is more of a framework optimization and there is no target side since it only operates on the host.